### PR TITLE
feat(ios): nbstore swift native binding

### PR DESCRIPTION
**Pull request by Brooooooklyn**

### CODE REVIEW

**Comment by Brooooooklyn** [comment]
* **#9211** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/9211?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/9211?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `canary`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/toeverything?ref=merge-queue-instructions-in-stack-comment&prId=9211).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

**Comment by Brooooooklyn** [comment]
### Merge activity

* **Dec 25, 10:34 AM EST**: [Graphite](https://app.graphite.dev/github/pr/toeverything/AFFiNE/9211) disabled "merge when ready" on this PR due to: a merge conflict with the target branch; resolve the conflict and try again..
* **Dec 27, 12:48 AM EST**: A user added this pull request to the [Graphite merge queue](https://app.graphite.dev/merges?org=toeverything&repo=AFFiNE).
* **Dec 27, 1:50 AM EST**: A user  merged this pull request with the [Graphite merge queue](https://app.graphite.dev/merges?org=toeverything&repo=AFFiNE).

File: packages/frontend/native/nbstore/src/lib.rs [hunk 1]
```diff
@@ -1,27 +1,47 @@
...
 use chrono::NaiveDateTime;
 use napi::bindgen_prelude::*;
 use napi_derive::napi;
 
-fn map_err(err: sqlx::Error) -> napi::Error {
-  napi::Error::from(anyhow::Error::from(err))
+#[cfg(feature = "noop")]
+type Result<T> = anyhow::Result<T>;
+
+#[cfg(not(feature = "noop"))]
```
Comment (EYHN) [inline packages/frontend/native/nbstore/src/lib.rs]: need a better feature name

File: packages/frontend/mobile-native/src/lib.rs [hunk 2]
```diff
@@ -1,8 +1,883 @@
...
+      size: blob.size,
+      mime: blob.mime,
+      created_at: blob.created_at.and_utc().into(),
+    }
+  }
+}
+
+#[derive(uniffi::Object)]
+pub struct DocStoragePool {
+  inner: DashMap<String, DocStorage>,
```
Comment (EYHN) [inline packages/frontend/mobile-native/src/lib.rs]: how about just store storage::SqliteDocStorage here, i think we dont need DocStorage

**Review by EYHN — APPROVED** [review]

**Pull request merged**